### PR TITLE
Add custom transforms to synchronous transform and swap to synchronous in js tests

### PR DESCRIPTION
**Pull request by peter-hamilton**
Only the async `transform(...)` was applying custom transforms. Now both do.
In the short term, js tests will use synchronous transform operation. This should be expanded later.

### CODE REVIEW

File: js/tests/swcify.test.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ describe('swcify', () => {
 });
 
 describe('async transform', () => {
```
Comment (TheMallen) [inline js/tests/swcify.test.ts:67]: this still says `async` but its testing sync now (like I said in a comment we might want to just have one for each though)

File: src/transform.rs [hunk 2]
```diff
@@ -14,121 +15,128 @@ use swc::{try_with_handler, Compiler, TransformOutput};
...
+                FileName::Anon
+            } else {
+                FileName::Real(options.filename.clone().into())
+            },
+            src,
+        ))
+    })
+}
+
+fn custom_transforms() -> (impl Fold, impl Fold) {
```
Comment (TheMallen) [inline src/transform.rs:138]: 👍  I like this, we can add the config wrangling stuff here down the line too :)

**Review by TheMallen — APPROVED** [review]
Looks good but we might have different formatters or something since it looks like saving files is reformatting them in opposite ways for each of us (not a problem for this PR). 

I like the division of the custom plugin stuff into a little function :)

File: js/tests/swcify.test.ts (hunk 1 above)
Comment (peter-hamilton) [reply js/tests/swcify.test.ts]: Clarified the description a little.

**Comment by peter-hamilton** [comment]
> > Synchronous transform operations are the main use case so the js tests were switched.
> 
> They are for jest but the webpack loader will be using async, confusing I know. 😓
> 
> Maybe we should have at least one test for each (we can have top level describes for both functions).

Oops. I added a single basic async test for now. To be expanded later.

**Pull request merged**